### PR TITLE
Pre release fixes

### PR DESCRIPTION
**Pull request by edlu77**
Issue - # (Put issue number preceeded by hash sign to link to issue)

Description of changes - 

@mentions of the person or team responsible for reviewing proposed changes.

TO-DO checklist before submitting your pull-request - 

- [ ] Are all acceptance criteria met ? 
- [ ] Has the code been documented properly ?
- [ ] Does component css-class structure conform to the Blue Dino style template i.e. top-level element .wrapper, .container and other inside css classes ?
- [ ] Have all checks passed ?
- [ ] Have you tested inter-browser functionality (should work on the latest Chrome, Firefox and Edge browsers) ?

### CODE REVIEW

**Comment by bherr2** [comment]
Wow, 14 code smells!?

**Comment by edlu77** [comment]
> Wow, 14 code smells!?

It's just a couple files, fixing them now.

File: src/app/core/header/header.component.html [hunk 1]
```diff
@@ -1,14 +1,17 @@
 <mat-toolbar class="ccf-header">
-  <mat-icon class="logo" aria-hidden="false" aria-label="Reset tool" matTooltip="Human BioMolecular Atlas Project"
-            matTooltipPosition="right" svgIcon="app:logo" (click)="logoClicked.emit()"></mat-icon>
-
+  <a class="portal-home" href="dataSourceService.dbOptions.hubmapDataUrl">
```
Comment (axdanbol) [inline src/app/core/header/header.component.html:2]: This href seems wrong? Shouldn't it be a reference to the hubmap portal page? @bherr2 Can you shed some light on this?

File: src/app/core/header/header.component.spec.ts [hunk 2]
```diff
@@ -17,12 +17,6 @@ describe('HeaderComponent', () => {
     expect(outputs.downloadClicked.emit).toHaveBeenCalled();
   });
 
-  it('should emit logoClicked when logo is clicked', async () => {
```
Comment (axdanbol) [inline src/app/core/header/header.component.spec.ts:20]: Replace this with a new test for the refresh button!

File: src/app/core/header/header.component.ts [hunk 3]
```diff
@@ -13,15 +13,18 @@ import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from
 })
 export class HeaderComponent {
 
+  constructor(readonly dataSourceService: DataSourceService) {
```
Comment (axdanbol) [inline src/app/core/header/header.component.ts:16]: Most likely there is no reason to have this injected (unless I'm wrong about the href in the html file)

File: src/app/core/models/image-viewer-layer.ts [hunk 4]
```diff
@@ -39,4 +39,49 @@ export class ImageViewerLayer {
...
   }
+
+}
+
+/**
+ * Converts a hex color code to RGB and calculates a value for the color brightness
+ * @param color Hex color code
+ * @returns brightness value
+ */
+function getBrightness(color: string): number {
```
Comment (axdanbol) [inline src/app/core/models/image-viewer-layer.ts:50]: Move these into the ImageViewerLayer class. You'll have to update the constructor argument by adding these to the `Omit<..., 'background' | 'getBrightness' | ...>`

File: src/app/modules/color-scheme/color-picker-launcher/color-picker-launcher.component.ts [hunk 5]
```diff
@@ -24,7 +24,28 @@ export class ColorPickerLauncherComponent {
    */
   @Output() layerChange = new EventEmitter<ImageViewerLayer>();
 
+  /**
+   * Class name for the color icon
+   */
   get className(): string {
```
Comment (axdanbol) [inline src/app/modules/color-scheme/color-picker-launcher/color-picker-launcher.component.ts:30]: This seems unused? If so remove.

File: src/app/modules/color-scheme/color-picker-launcher/color-picker-launcher.component.ts [hunk 6]
```diff
@@ -24,7 +24,28 @@ export class ColorPickerLauncherComponent {
...
     return `color-icon ${this.layer.colorScheme.type === 'discrete' ? 'circle' : ''}`;
   }
+
+  /**
+   * Determines whether the color icon is light enough to require a border (light mode only)
+   * @param color The color hex code
+   * @returns true if brightness is above a certain threshold
+   */
+  iconLight(color: string): boolean {
```
Comment (axdanbol) [inline src/app/modules/color-scheme/color-picker-launcher/color-picker-launcher.component.ts:39]: These two functions can be removed once you've moved the isLight/isDark onto the ImageViewerLayer class

File: src/app/modules/color-scheme/color-scheme-contents/color-scheme-contents.component.scss [hunk 7]
```diff
@@ -9,8 +9,8 @@
   flex-direction: column;
   position: absolute;
   z-index: 1;
-  right: -1rem;
-  top: -1rem;
+  right: calc(-1rem - 2px);
```
Comment (axdanbol) [inline src/app/modules/color-scheme/color-scheme-contents/color-scheme-contents.component.scss:12]: `1rem = 16px` -> `2px = 0.125rem` -> `-1rem - 2px = -1.125rem`
No calc needed!

File: src/app/modules/image-viewer/image-viewer-legend/image-viewer-legend.component.ts [hunk 8]
```diff
@@ -15,4 +15,22 @@ export class ImageViewerLegendComponent {
    * The list of layers used to render this components
    */
   @Input() layers: ImageViewerLayer[];
+
+  /**
+   * Determines whether the color icon is light enough to require a border (light mode only)
+   * @param color The color hex code
+   * @returns true if brightness is above a certain threshold
+   */
+  iconLight(color: string): boolean {
```
Comment (axdanbol) [inline src/app/modules/image-viewer/image-viewer-legend/image-viewer-legend.component.ts:24]: Again these can be removed when is(Light|Dark) has been placed inside the ImageViewerLayer class

File: src/app/shared/components/drawer/toggle-button/toggle-button.component.scss [hunk 9]
```diff
@@ -4,22 +4,12 @@
   right: calc(-1rem - 1px);
```
Comment (axdanbol) [inline src/app/shared/components/drawer/toggle-button/toggle-button.component.scss:4]: No need for calc! Remember `1rem = 16px`! Convert to rems

File: src/app/shared/components/spinner-overlay/spinner-overlay.component.scss [hunk 10]
```diff
@@ -8,7 +8,7 @@
   flex-direction: column;
   justify-content: center;
   align-items: center;
-  background-color: black;
+  background-color: #FAFAFA;
```
Comment (axdanbol) [inline src/app/shared/components/spinner-overlay/spinner-overlay.component.scss:11]: This should be moved to the theming file

File: src/styles.scss [hunk 11]
```diff
@@ -29,11 +29,11 @@ body {
 }
 
 body {
+  @include theming.ccf-theme(light.$theme);
```
Comment (axdanbol) [inline src/styles.scss:32]: This should be inside a .light-theme class just like dark theme

File: src/app/core/header/header.component.html (hunk 1 above)
Comment (bherr2) [reply src/app/core/header/header.component.html:2]: Yes, it should be. Depending on the environment, the portal url will be different, thus it needs to be injected. It shouldn't be hubmapDataUrl, but hubmapPortalUrl -- ie https://github.com/hubmapconsortium/ccf-ui/blob/develop-v2/projects/ccf-database/src/lib/ccf-database.ts#L25

File: src/app/core/header/header.component.ts (hunk 3 above)
Comment (bherr2) [reply src/app/core/header/header.component.ts:16]: That's where the hubmapPortalUrl is injected from, but this should probably be provided via an @Input from app.component

Comment (bherr2) [reply src/app/core/header/header.component.ts:16]: @Input() hubmapPortalUrl: string

**Pull request merged**